### DISCUSSION TITLE
Chat Memory API in Spring AI

### DISCUSSION THREAD

**Discussion by iAMSagar44** [spring-projects/spring-ai#391] [General]
Hi,

Is there a functionality in SpringAI to maintain the chat message history, like the ChatMemory class in LangChain4j?
[Chat Memory](https://docs.langchain4j.dev/tutorials/chat-memory)

**Comment by safaahashim** [comment]
The memory feature seems essential for any serious LLM application that implements a chatbot or other conversational AI application. Hope we get some update/clarification if this feature is on the roadmap.

**Comment by iamroberthanson** [comment]
If anyone is working on an implementation of memory support I would be interested in seeing what your interfaces look like.

**Comment by ThomasVitale** [comment]
It looks like it's in progress: https://github.com/spring-projects/spring-ai/issues/396#issuecomment-1991774898

**Comment by shandilya07** [comment]
I don't there is something which is implicitly provided as part of the Spring AI features, however, I recently did implement some sort of makeshift memory for the chat bot using Open AI + Spring AI. If you are interested, you could check this video I made on this topic : https://www.youtube.com/watch?v=83KTwxCCe9I
